core/drivers/crypto/caam/include/caam_io.h

```diff
@@ -20,6 +20,11 @@
 /* Big Endian 32 bits Value access */
 #define caam_read_val32(a)	get_be32(a)
 #define caam_write_val32(a, v)	put_be32(a, v)
+
+#define caam_get64(a)	(						\
+		((uint64_t)get_be32(a) << 32) | \
+		(get_be32((void *)((uintptr_t)(a) + 4))))
+
```

Comment:
The following construct should fix the checkpatch warnings:
```C
#define caam_get64(a)                               \
({                                                  \
        typeof(a) _a = (a);                         \
        ((uint64_t)get_be32(_a) << 32) |            \
        (get_be32((void *)((uintptr_t)(_a) + 4)))); \
})
```

And same below of course.

Reply:
I was about to suggest an inline function instead.

Reply:
Yep even better ;-)

Reply:
I have made the changes. Can you please suggest which checkpatch does this repository use? As I was not able to find this failure in the Linux version 5.4 and checkpatch version 0.32

Reply:
How about something like:
``` C
static inline uint64_t caam_get64(const uint64_t *a)
{
        const uint32_t *a32 = (const uint32_t *a);

        return SHIFT_U64(get_be32(a32[0]), 32) | get_be32(a32[1]);
}
```

Reply:
@priyankasingh01 
What will be the usage of this function? If the function is to be a HW register access function, please name it io_read_val64 and it's to get a caam decriptor value, name it caam_read_val64.
Other point is concerning the return value. paddr_t is 32 bits on 32 bit architecture and 64 bits on 64 bit architecture, hence I would advice to use a uint64_t return instead.

Reply:
Hi @cneveux ,
This function is for reading the output ring entries of CAAM job ring.
In case of 64bit architecture the output ring entries will be like
|   64bit desc addr   |   32 bit status |    64 bit desc addr   | 32 bit status | and so on.

So if we use the function you wrote, we can read the first entry successfully, but in case of reading the second entry the address is not 64 bit aligned and when we try to read it as 64 bit it results in alignment fault, thats why we are reading this as 32 bit values.

Regarding paddr_t , I agree with you, will change it to uint64_t.